pss/pss.go

```diff
@@ -553,22 +556,58 @@ func (p *Pss) isSelfPossibleRecipient(msg *PssMsg, prox bool) bool {
 // SECTION: Message sending
 /////////////////////////////////////////////////////////////////////
 
-func (p *Pss) enqueue(msg *PssMsg) error {
+// Enqueue a message msg. Pending will be true for messages with a booked slot in the queue
+func (p *Pss) enqueue(msg *PssMsg, pending bool) error {
 	defer metrics.GetOrRegisterResettingTimer("pss.enqueue", nil).UpdateSince(time.Now())
 
 	outboxmsg := newOutboxMsg(msg)
-	select {
-	case p.outbox <- outboxmsg:
-		metrics.GetOrRegisterGauge("pss.outbox.len", nil).Update(int64(len(p.outbox)))
 
-		return nil
-	default:
+	p.outboxMutex.Lock()
+	defer p.outboxMutex.Unlock()
+	pendingSize := p.getPending()
+	// Only allow defaultOutboxCapacity messages at most processed (both enqueued or being forwarded)
```

Comment:
Changed in commit f73157b

Reply:
It sounds good to me. 
THe only thing is that we only implemented the mechanism to ensure that a reenqued message always have a slot booked in the outbox, we didn't implement parallel processing of messages.
I think that in order to test both solutions we shoyuld compare them with parallel processing in both cases.
Anyway, I will implement @zelig proposal (with the augmented buffer to avoid deadlocks) in a different branch to compare them.

Reply:
Great. Should we change the label to `in progress` again maybe?

Reply:
In deed. Until we can't compare perfromance with the other solution it should be in progress.